Content.Server/GameObjects/Components/Access/AccessReaderComponent.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Generic;
+using Content.Server.Interfaces.GameObjects;
+using Content.Shared.GameObjects.Components.Inventory;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Access
+{
+    [RegisterComponent]
+    public class AccessReader : Component
+    {
+        public override string Name => "AccessReader";
+        private List<string> _necessaryTags;
+        private List<string> _sufficientTags;
+
+        public bool IsAllowed(IEntity entity)
+        {
+            var accessProvider = FindAccessProvider(entity);
+            return accessProvider != null && IsAllowed(accessProvider);
+        }
+
+        private bool IsAllowed(AccessComponent accessProvider)
+        {
+            foreach (var sufficient in _sufficientTags)
+            {
+                if (accessProvider.Tags.Contains(sufficient))
+                {
+                    return true;
+                }
+            }
+            foreach (var necessary in _necessaryTags)
+            {
+                if (!accessProvider.Tags.Contains(necessary))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        [CanBeNull]
+        private static AccessComponent FindAccessProvider(IEntity entity)
+        {
+            if (entity.TryGetComponent(out AccessComponent accessComponent))
```

Comment:
I don't know if this will ever be a good idea. I was thinking of things like cyborgs when I wrote it.

Reply:
I don't see an immediate problem with it.